go/tools/builders/nogo_vet.go

```diff
@@ -111,16 +110,27 @@ func buildVetcfgFile(packageFile, importMap map[string]string, stdImports, files
 		vcfg.Standard[imp] = true
 	}
 
-	js, err := json.MarshalIndent(vcfg, "", "\t")
+	// GRIPE: vet checks whether its first positional argument has the suffix
+	// "vet.cfg", rather than accepting the file as an option.
+	vcfgFile, err := ioutil.TempFile("", "nogo-*-vet.cfg")
 	if err != nil {
-		return "", fmt.Errorf("internal error marshaling vet config: %v", err)
-	}
-	js = append(js, '\n')
-	vcfgFile := filepath.Join(os.TempDir(), "vet.cfg")
-	if err := ioutil.WriteFile(vcfgFile, js, 0666); err != nil {
 		return "", err
 	}
-	return vcfgFile, nil
+	vcfgPath := vcfgFile.Name() // vcfgPath may be "" when returning
+	defer func() {
+		if cerr := vcfgFile.Close(); err == nil && cerr != nil {
+			os.Remove(vcfgPath)
```

Comment:
Do you want to `os.Remove` on any error, not just an error closing the file, so you always clean up the temp file?

So

```
defer func() {
  if cerr := vcfgFile.Close(); err == nil && cerr != nil {
    err = cerr
  }
  if err != nil {
    os.Remove(vcfgPath)
  }
```

(I think)

Reply:
Good catch, it should be on any error.